libraries/SPI/src/SPI.h

```diff
@@ -144,26 +158,90 @@ class SPIClass {
     void begin(void);
     void end(void);
 
-    // Attach/Detach pin to/from SPI controller
+    // Attach CS pin to SPI instance
     void begin(uint8_t _pin);
-    void end(uint8_t _pin);
 
-    // These methods sets a parameter on a single pin
+    /* These methods sets a parameter on a single pin.
+     * These methods must be called after initialization of the SPI
+     * instance with begin() or beginTransaction().
+     * @NOTE
+     * These functions should not be used in new projects.
+     * Use SPISettings with SPI.beginTransaction() to configure SPI parameters.
+     */
     void setBitOrder(uint8_t _pin, BitOrder);
     void setDataMode(uint8_t _pin, uint8_t);
     void setClockDivider(uint8_t _pin, uint8_t);
 
-    // These methods sets the same parameters but on default pin BOARD_SPI_OWN_SS
-    void setBitOrder(BitOrder _order) { setBitOrder(BOARD_SPI_OWN_SS, _order); };
-    void setDataMode(uint8_t _mode) { setDataMode(BOARD_SPI_OWN_SS, _mode); };
-    void setClockDivider(uint8_t _div) { setClockDivider(BOARD_SPI_OWN_SS, _div); };
+    // These methods sets the same parameters for the default CS pin.
+    void setBitOrder(BitOrder _order) {
+      setBitOrder(CS_PIN_CONTROLLED_BY_USER, _order);
+    }
+    void setDataMode(uint8_t _mode) {
+      setDataMode(CS_PIN_CONTROLLED_BY_USER, _mode);
+    }
+    void setClockDivider(uint8_t _div) {
+      setClockDivider(CS_PIN_CONTROLLED_BY_USER, _div);
+    }
 
   private:
-    //uint32_t id;
-    SPISettings spiSettings[SPI_CHANNELS_NUM+1];
-    int8_t g_active_id;
-    uint8_t _pin;
-    spi_t _spi;
+    spi_settings_t  spiSettings[NB_SPI_SETTINGS];
+    int16_t         _CSpin;
+    spi_t           _spi;
+
+    uint8_t getIdx(uint8_t _pin)
```

Comment:
Merge getIdx and addIdx and use a second arg to define if a new idx could be added with a default value (no add by default)
This will avoid to loop twice if there is no idx for pin then call addIdx.